### DISCUSSION TITLE
Documentaion Improvement and feature request

### DISCUSSION THREAD

**Discussion by PureOxygen-Dev** [Weatherlights/Winget-AutoUpdate-Intune#51] [Ideas]
Hi @Weatherlights 

Let me explain a little the issues we were facing with the tooling\documentation. 

Let's use Adobe Acrobat as the example.  We need to do three post install events

Adobe.Acrobat.Reader.32-bit-installed {"AddModsReg": [ { "AddKey": "HKLM:\\SOFTWARE\\Policies\\Adobe\\Acrobat Reader\\DC\\FeatureLockdown", "AddValue": "bAcroSuppressUpsell", "AddTypeData": "1", "AddType": "Dword"}]}

Adobe.Acrobat.Reader.32-bit-installed {"RemoveModsLnk" : [{"Shortcut":"Acrobat Reader"]}

Adobe.Acrobat.Reader.32-bit-installed {"RemoveModsLnk" : [{"Shortcut":"Adobe Acrobat"]}

I think in the documentation you might call out all the available functions and the required values to call them.

Also you could explain that to call multiple actions you do this by building a well-formed JSON as below and srinking it to one line.

{
  "AddModsReg": [
    {
      "AddKey": "HKLM:\\SOFTWARE\\Policies\\Adobe\\Acrobat Reader\\DC\\FeatureLockdown",
      "AddValue": "bAcroSuppressUpsell",
      "AddTypeData": "1",
      "AddType": "Dword"
    }
  ],
  "RemoveModsLnk": [
    {"Shortcut": "Adobe Acrobat"},
    {"Shortcut": "Acrobat Reader"}
  ]
}

As Powershell developers we were able to look at the code and figure this out, but some people using this tool might need a little more documentation.

_Question Feature Request_
Do you have a roadmap for complex self code powershell to be able to be pushed in the mods?

Allowing for admins to be able to extend the capabilities missing from the available functions you offer in the single line JSON examples.

**Comment by Weatherlights** [comment]
Hey

You are right regarding the documentation it could need an update here and I hope I can do this now during the holidays. Of course you are invited to add some documentation aswell ;).

Regarding complex self code: Maybe something that can be used within the JSON code... so maybe something like this would come to my mind:
```
"PowershellMod": [
{
"Script Code"
}]
```
What do you think? Would that help you?

Regarding complex scripts that are file based: Since code integrity is a part of my solution executing entire scripts would break with the approach. One idea is that WAUaaS will work even when Powershell Execution Policy is set to AllSigned or Defender Application Control is used.

P.s.: If you need to modify configurations in ini-, xml- or json- you can also take a look at the [PolicyApplicator](https://policyapplicator.weatherlights.com) project of mine. This tool allows you to configure applications regardless of the installation by using Intune custom profiles. It can also capture existing configurations and convert them into ready to use profiles :).
